### ISSUE TITLE
Scrape "Claimed At" time for CNN

### ISSUE THREAD

**Issue by slifty** [TechAndCheck/tech-and-check-alerts#100]
We want to try to get the time a claim was made (resolution of program start time is fine for now, but as specific as possible given the transcript is what we should ultimately aim for).

This will be a matter of finding the date + time on the transcript page; worst case we can use the date in the URL but that will only give us resolution of the day, not the hour.

**Comment by reefdog** [comment]
Quick thought that we'll want to make sure we sniff out the timezone for the timestamp, too, and if we're doing anything like the worst case (URL date) that we're localizing it to UTC and considering how that will affect our queries.

**Comment by slifty** [comment]
So!  Twitter this should be really easy since I think twitter does UTC.

CNN, however, seems to write "ET" which isn't even an actual time zone.  I'm honestly a little flummoxed as to what to do, especially since historic scrapes make the problem 2x harder (since EST / EDT will change based on the day) AND we don't actually care about historic scrape.

Options are:

1) Convert to UTC based on what the time would be if it was aired at the "ET" time *today*
2) Try to be smarter to figure out whether it is EST or EDT based on the air date.

#2 is the more correct option, but I'm not sure it is worth the effort it will take.

**Comment by reefdog** [comment]
I think you mean (B) is the more correct but harder option? (And assuming we're converting to / storing UTC in both options, anyway.)

I can't believe everyone isn't using a `<time>` element with a valid ISO 8601 `datetime` attribute. Tsk tsk.

**Comment by slifty** [comment]
I suppose I do mean B if you want to be exact about it.

**Comment by reefdog** [comment]
If at all possible.

**Comment by slifty** [comment]
OK so I spent some time yesterday digging into this -- yikes what a nightmare.

Some things worth putting down as a baseline (which folks probably already know):

1. Javascript's native Date object does not do time zones
2. `dayjs` unfortunately does not do time zones
3. `moment` does, but it's so huge (RIP)
4. [Luxon](https://moment.github.io/luxon/index.html) is a spinoff of moment which is much smaller and also has timezone support.
5. There's this [spacetime](https://github.com/spencermountain/spacetime) thing which is explicitly for moving between time zones.

I'm not sure we can get around the need for another third party at this point.  Eventually it looks like [dayjs will have time zone support](https://github.com/iamkun/dayjs/issues/46) but it isn't there yet.

I'm not sure the value of dayjs if we have to use another library to supplement it.

**Comment by reefdog** [comment]
spacetime is big (15.5KB minzipped) but let's use it until dayjs gains timezones.

```js
const preDST = spacetime('2019-01-01T10:00', 'America/New_York')
preDST.format('nice')
// "Jan 1st, 10:00am"
preDST.goto('UTC').format('nice')
// "Jan 1st, 3:00pm"

const postDST = spacetime('2019-08-01T10:00', 'America/New_York')
postDST.format('nice')
// "Aug 1st, 10:00am"
postDST.goto('UTC').format('nice')
// "Aug 1st, 2:00pm"
```

We'll want to clean up and/or reformat the dates we pass into `spacetime()` and test to make sure it worked, since spacetime will happily receive any garbage and just default to Unix epoch.

**Comment by reefdog** [comment]
(Copying in a convo from Slack. )

@slifty:
> Anyway i dont plan to pass strings to it but rather will use dayjs for initial parsi g

My head is getting into a scramble about whether it’s safe to pass the string through dayjs for reformatting. I guess so? I guess it’s immaterial what timezone dayjs _thinks_ it’s operating in, so long as it spits out something with the original values (reformatted) and no timezone?

**Comment by slifty** [comment]
Yeah -- dayjs just works in essentially absolute time, not caring about time zones whatsoever.  It's really a library for comparing, parsing, and formatting dates.  I think it is fine to use that as the primary date vector and use spacetime purely as a utility for transitioning between time zones (again, until dayjs just supports that)

**Comment by reefdog** [comment]
Quick note surfaced by #181: CNN sometimes typos the dates in the transcript, so once we're scraping `claimedAt` from the CNN content, we'll want to set some sensible parameters around it like "if `claimedAt` is in the future, fall back to today and log a warning" or something.
